src/index.js

```diff
@@ -26,10 +30,32 @@ type Options = {
 };
 */
 
-export default function prepass(
+let onDiff;
+let onRender;
+
+function prepass(
+	vnode /*: VNode */,
+	visitor /*: ?(vnode: VNode, component: typeof Component) => ?Promise<any> */,
+) /*: Promise<any|Array<any>> */ {
+	const prevVnodeHook = options.vnode;
+	const prevSkipEffects = options[_skipEffects];
+
+	onDiff = options[_diff]
+	onRender = options[_render]
+	options.vnode = undefined;
```

Comment:
Why do we clear the vnode hook? Does this still work when used with compat?

Reply:
The VNode hook only affects dom nodes in current compat hence my thinking was to avoid us paying twice due to prepass --> RTS we have three choices

- we combine prepass and RTS in an async RTS
- we deprecate prepass and point people at straming
- we avoid paying for options.vnode